src/test/webapp/app/job/my-positions/my-positions-page.component.spec.ts

```diff
@@ -0,0 +1,292 @@
+    expect(loadSpy).toHaveBeenCalled();
+  });
+
+  it('should handle close job error', async () => {
+    vi.spyOn(jobService, 'changeJobState').mockReturnValueOnce(throwError(() => new Error('close fail')));
+    await component.onCloseJob('1');
+    expect(toastService.showError).toHaveBeenCalledWith({
+      detail: expect.stringContaining('close fail'),
+    });
+  });
+
+  it('should correctly build stateTextMap from availableStatusOptions', () => {
+    const map = component.stateTextMap();
+    expect(map.DRAFT).toBe('jobState.draft');
+    expect(map.PUBLISHED).toBe('jobState.published');
+    expect(map.CLOSED).toBe('jobState.closed');
+    expect(map.APPLICANT_FOUND).toBe('jobState.applicantFound');
+  });
+
+  it('should return early from loadJobs when userId is empty', async () => {
+    vi.spyOn(accountService, 'loadedUser').mockReturnValue({
+      id: '',
+      email: '',
+      name: '',
+    });
+    const spy = vi.spyOn(jobService, 'getJobsByProfessor');
+    await (component as unknown as { loadJobs: () => Promise<void> }).loadJobs();
+    expect(spy).not.toHaveBeenCalled();
+  });
+
+  it('should handle loadOnTableEmit when event.first and rows are undefined', async () => {
+    const spy = vi.spyOn(component as unknown as { loadJobs: () => Promise<void> }, 'loadJobs').mockResolvedValue();
+    component.pageSize.set(5);
+    component.loadOnTableEmit({});
+    expect(component.page()).toBe(0);
+    expect(component.pageSize()).toBe(5);
+    expect(spy).toHaveBeenCalled();
+  });
+
+  it('should fallback to original key when translation key not found in mapTranslationKeysToEnumValues', () => {
+    const result = (
+      component as unknown as { mapTranslationKeysToEnumValues: (keys: string[]) => string[] }
+    ).mapTranslationKeysToEnumValues(['unknownKey']);
+    expect(result).toEqual(['unknownKey']);
+  });
+
+  it('should set userId to empty string when loadedUser returns undefined', async () => {
+    vi.spyOn(accountService, 'loadedUser').mockReturnValue(undefined);
+    const spy = vi.spyOn(jobService, 'getJobsByProfessor');
+    await (component as unknown as { loadJobs: () => Promise<void> }).loadJobs();
+    expect(component.userId()).toBe('');
+    expect(spy).not.toHaveBeenCalled();
+  });
+
+  it('should default jobs and totalRecords when API returns undefined content and totalElements', async () => {
+    vi.spyOn(accountService, 'loadedUser').mockReturnValue({
+      id: 'u1',
+      email: '',
+      name: '',
+    });
+    jobService.getJobsByProfessor.mockReturnValueOnce(
+      of({ content: undefined, totalElements: undefined }) as unknown as ReturnType<typeof jobService.getJobsByProfessor>,
+    );
+    await (component as unknown as { loadJobs: () => Promise<void> }).loadJobs();
+
+    expect(component.jobs()).toEqual([]);
+    expect(component.totalRecords()).toBe(0);
+  });
+
+  it('should call loadJobs with non-empty filters', async () => {
+    accountService.loadedUser.mockReturnValue({
```

Comment:
```suggestion
    vi.spyOn(accountService, 'loadedUser').mockReturnValue({
```

Sometimes you would use `vi.spyOn()` but in this case a direct mock method.